src/custom-style-interface.js

```diff
@@ -105,11 +107,51 @@ export default class CustomStyleInterface {
     }
     return cs;
   }
+  /**
+   * @param {!HTMLStyleElement} style
+   * @return {boolean}
+   */
+  isDocumentStyle(style) {
+    return !style.hasAttribute('scope');
+  }
+  gatherMainDocumentStyles() {
+    const styles = document.querySelectorAll('style:not([scope])');
+    for (let i = 0; i < styles.length; i++) {
+      const s = /** @type {!HTMLStyleElement} */(styles[i]);
+      this.addCustomStyle(s);
+    }
+  }
+  watchMainDocumentStyles() {
+    if (this.observer) {
+      return;
+    }
+    this.gatherMainDocumentStyles();
+    /**
+     * @param {Array<MutationRecord>} mxns
+     */
+    let mutationHandler = (mxns) => {
+      for (let i = 0; i < mxns.length; i++) {
+        let mxn = mxns[i];
+        for (let j = 0; j < mxn.addedNodes.length; j++) {
+          let n = mxn.addedNodes[j];
+          if (n.nodeType === Node.ELEMENT_NODE && n.localName === 'style') {
+            const el = /** @type {!HTMLStyleElement} */(n);
+            if (this.isDocumentStyle(el)) {
+              this.addCustomStyle(el);
```

Comment:
Add a comment here noting that `addCustomStyle` dedups so we don't have to worry about add/remove/add before validation.

Reply:
Yep, will do!